app/components/edit/BuyRouteEdit.tsx

```diff
@@ -53,6 +53,9 @@ const BuyRouteEdit = ({
     const buyTypePreselection = getBuyTypePreselection();
     if (buyTypePreselection) setValue("type", buyTypePreselection);
 
+    const blockchains = getBlockchains();
+    if (blockchains.length === 1) setValue("blockchain", blockchains[0]);
+
```

Comment:
I'm sorry, we introduced the bug due to my wrong comment 😞  I would remove this again and adapt `getBuyableAssets`:
```ts
  const getBuyableAssets = (values: Asset[]): Asset[] => {
    return values.filter((a) => (!blockchain || a.blockchain === blockchain) && a.buyable);
  };
```